lib/wallet/wallet.js

```diff
@@ -3468,6 +3468,24 @@ class Wallet extends EventEmitter {
     if (!account)
       return null;
 
+    // The account index in the db may be wrong.
+    // We must read it from the stored xpub to be
+    // sure of its correctness.
+    //
+    // For more details see:
+    // https://github.com/bcoin-org/bcoin/issues/698.
+    //
+    // TODO(boymanjor): remove index manipulation
+    // once the watch-only wallet bug is fixed.
+    account.accountIndex = account.accountKey.childIndex;
+
+    // Unharden the account index, if necessary.
+    if (account.accountIndex & HD.common.HARDENED)
+      account.accountIndex ^= HD.common.HARDENED;
+
+    // Update the path's account index as well.
+    path.account = account.accountIndex;
```

Comment:
Is this necessary? It looks like `Account.derivePath()` only reads `branch` and `index` from the `path` argument (in this case `index` is the final bip32 field, not the account index).

https://github.com/handshake-org/hsd/blob/fc4623d8750399ff086e6de8c023e6dd8c5aa39d/lib/wallet/account.js#L388-L392

and then `Account.deriveKey()` doesn't need anything else from `path`...

Reply:
We discussed this comment out of band. TL;DR - it is necessary because the wallet uses its current `accountDepth` for the account index of newly created accounts. This behavior breaks if an xpub is imported, which encodes an account index different from the wallet's current `accountDepth`.

Reply:
I understand that but why update `path.account`? I don't see where that value is ever checked and if I comment out that line, my ad hoc test still passes, as does the one you added in 5611feb18687b5b047621804b51554c19eff6e0e

Reply:
Ah, I see. I've removed the unnecessary update.